aea/cli/ipfs_hash.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script generates the IPFS hashes for all packages.
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+import collections
+import csv
+import operator
+import os
+import re
+import sys
+import traceback
+from pathlib import Path
+from typing import Collection, Dict, List, Optional, Tuple, Type, cast
+
+import click
+
+from aea.configurations.base import (
+    AgentConfig,
+    ConnectionConfig,
+    ContractConfig,
+    PackageConfiguration,
+    PackageType,
+    ProtocolConfig,
+    SkillConfig,
+    _compute_fingerprint,
+)
+from aea.configurations.loader import ConfigLoaders
+from aea.helpers.ipfs.base import IPFSHashOnly
+from aea.helpers.yaml_utils import yaml_dump, yaml_dump_all
+
+
+HASHES_FILE = "hashes.csv"
+HASH_TOOL = IPFSHashOnly()
```

Comment:
Is this good practice? The object will be instantiated when the library is loaded. It can potentially lead to confusing behaviour.